src/main/register-events.ts

```diff
@@ -21,11 +30,69 @@ export default function registerAllEvents(window: BrowserWindow | null) {
 
     ipcMain.on('hw-status', (event) => {
         event.reply('hw-change', HWPing.Instance.currentStatus);
-    })
+    });
+
+    ipcMain.on('steps:set', (_, [step]) => {
+        store.set('currentStep', step);
+        if (store.get('stepsStartedAt') === 0) {
+            store.set('stepsStartedAt', new Date().getTime());
+        }
+    });
+
+    ipcMain.on('steps:get', (event) => {
+        // This is the step that we'll reply with
+        let stepToReply = 0;
+
+        // Get the step start date
+        const lastStart = store.get('stepsStartedAt');
+
+        // If we don't have a date key (I.E. it's 0), we haven't started the steps yet
+        if (!lastStart) {
+            stepToReply = 0;
+        } else {
+            // Convert to date
+            const startDate = new Date(lastStart);
+
+            // If today is Monday, and the start date is not today, reset the steps
+            const todayIsMonday = new Date().getDay() === 1;
+            const startDateIsToday =
+                startDate.toDateString() === new Date().toDateString();
+
+            if (todayIsMonday && !startDateIsToday) {
```

Comment:
The carts will basically never be powered on on Monday, we'll probably need to do like "if a Monday has passed since the startDate, reset everything"

Reply:
Classic blunder. Fixed in cdea532